mm2src/coins/nft/trading_proto_v2/mod.rs

```diff
@@ -0,0 +1,565 @@
+            Token::FixedBytes(args.maker_secret_hash.to_vec()),
+            Token::Uint(U256::from(time_lock)),
+        ])
+    }
+
+    /// ERC721 contract has overloaded versions of the `safeTransferFrom` function,
+    /// but `Contract::function` method returns only the first if there are overloaded versions of the same function.
+    /// Provided function retrieves the `safeTransferFrom` variant that includes a `bytes` parameter.
+    /// This variant is specifically used for transferring ERC721 tokens with additional data.
+    fn erc721_transfer_with_data(&self) -> Result<&ethabi::Function, Erc721FunctionError> {
+        let functions = ERC721_CONTRACT
+            .functions_by_name("safeTransferFrom")
+            .map_err(|e| Erc721FunctionError::AbiError(ERRL!("{}", e)))?;
+
+        // Find the correct function variant by inspecting the input parameters.
+        let function = functions
+            .iter()
+            .find(|f| {
+                f.inputs.len() == 4
+                    && matches!(
+                        f.inputs.last().map(|input| &input.kind),
+                        Some(&ethabi::ParamType::Bytes)
+                    )
+            })
+            .ok_or_else(|| {
+                Erc721FunctionError::FunctionNotFound(
+                    "Failed to find the correct safeTransferFrom function variant".to_string(),
+                )
+            })?;
+        Ok(function)
+    }
+
+    async fn validate_from_to_and_maker_status(
+        &self,
+        tx_from_rpc: &Web3Tx,
+        expected_from: Address,
+        expected_to: Address,
+        maker_status: U256,
+    ) -> ValidatePaymentResult<()> {
+        if maker_status != U256::from(MakerPaymentStateV2::PaymentSent as u8) {
+            return MmError::err(ValidatePaymentError::UnexpectedPaymentState(format!(
+                "NFT Maker Payment state is not PAYMENT_STATE_SENT, got {}",
+                maker_status
+            )));
+        }
+        if tx_from_rpc.from != Some(expected_from) {
+            return MmError::err(ValidatePaymentError::WrongPaymentTx(format!(
+                "NFT Maker Payment tx {:?} was sent from wrong address, expected {:?}",
+                tx_from_rpc, expected_from
+            )));
+        }
+        // As NFT owner calls "safeTransferFrom" directly, then in Transaction 'to' field we expect token_address
+        if tx_from_rpc.to != Some(expected_to) {
+            return MmError::err(ValidatePaymentError::WrongPaymentTx(format!(
+                "NFT Maker Payment tx {:?} was sent to wrong address, expected {:?}",
+                tx_from_rpc, expected_to,
+            )));
+        }
+        Ok(())
+    }
+
+    /// Retrieves the payment status from a given smart contract based on the swap ID and state type.
+    async fn payment_status_v2(
+        &self,
+        swap_contract: Address,
+        swap_id: Token,
+        contract: &Contract,
```

Comment:
we have `swap_contract` parameter and `contract` param, please provide good names to distinguish between the first which is an address and the second which is the contract abi.

Reply:
agree, done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2084/commits/23d1c994e15eab82cd47fdf880a7ef81e7c6091c